### ISSUE TITLE
Site fails if LocalStorage is explicitly disallowed with option useColorSchemeMediaQuery

### ISSUE THREAD

**Issue by c-kirkeby** [system-ui/theme-ui#553]
**Describe the bug**

**To Reproduce**
(Similar issues in both Firefox v72 and Chrome v79.0.3945.88 Windows 10 64-bit)
1. Go to https://gatsby.norse.codes
2. Go to your settings to block LocalStorage and Cookies on the site)
*Firefox:* about:preferences#privacy > Manage Permissions, add the site https://gatsby.norse.codes and click 'Block' and 'Save Changes'.
*Chrome:* chrome://settings/content/cookies > Add and enter the site gatsby.norse.codes then click 'Add'
3. Go to https://gatsby.norse.codes and refresh.
The error will differ depending on the browser but they both complain about not being able to access LocalStorage.

Firefox:
```
Request to access cookies or storage on “<URL>” was blocked because of custom cookie permission. 4
Request to access cookies or storage on “https://gatsby.norse.codes/blog” was blocked because of custom cookie permission.
2 blog:1250:8
Request to access cookies or storage on “https://gatsby.norse.codes/blog” was blocked because of custom cookie permission.
blog:2153:40
Request to access cookies or storage on “https://gatsby.norse.codes/blog” was blocked because of custom cookie permission.
blog:2:19
Request to access cookies or storage on “https://gatsby.norse.codes/blog” was blocked because of custom cookie permission.
blog:1:31821
DOMException: "The operation is insecure." d618faa047230c1c52d10b908bd3b5a512e1c154.js:1:381107
    fl https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    callback https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    vi https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    yi https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    vu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    unstable_runWithPriority https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Va https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    yu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    iu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Ya https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    unstable_runWithPriority https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Va https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Ya https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Qa https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    lu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Yu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    hydrate https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    UxWs https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
SecurityError: The operation is insecure. d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    H https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    G https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    hl https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    xu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    unstable_runWithPriority https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Va https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    ku https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    iu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    iu self-hosted:920
    Ya https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    unstable_runWithPriority https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Va https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Ya https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Qa https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    lu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    Yu https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    hydrate https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
    UxWs https://d33wubrfki0l68.cloudfront.net/bundles/d618faa047230c1c52d10b908bd3b5a512e1c154.js:1
```

Chrome:
```
bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1 DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
    at H (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:31839)
    at https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:32145
    at hl (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:381814)
    at xu (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:406540)
    at t.unstable_runWithPriority (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:272735)
    at Va (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:349499)
    at ku (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:406303)
    at iu (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:391243)
    at https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:349789
    at t.unstable_runWithPriority (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:272735)
fl @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
El.n.callback @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
vi @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
yi @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
vu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
t.unstable_runWithPriority @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Va @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
yu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
iu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
t.unstable_runWithPriority @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Va @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Ya @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Qa @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
lu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Yu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
hydrate @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
setTimeout (async)
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Promise.then (async)
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Promise.then (async)
UxWs @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
a @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
r @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
t @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1 Uncaught DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
    at H (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:31839)
    at https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:32145
    at hl (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:381814)
    at xu (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:406540)
    at t.unstable_runWithPriority (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:272735)
    at Va (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:349499)
    at ku (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:406303)
    at iu (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:391243)
    at https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:349789
    at t.unstable_runWithPriority (https://d33wubrfki0l68.cloudfront.net/bundles/bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1:272735)
Ya @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Qa @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
lu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Yu @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
hydrate @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
setTimeout (async)
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Promise.then (async)
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
Promise.then (async)
UxWs @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
a @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
r @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
t @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
(anonymous) @ bf3efeffc1e0f3164b6b5256cd39adbe09a477bb.js:1
5The resource <URL> was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
```

**Expected behavior**
The site renders fine with JavaScript disabled, and does not seem to need the `theme-ui-color-mode` key present when this is the case. I would hope that it would try something similar when Local Storage is disallowed, especially with [the `useColorSchemeMediaQuery: true` config option set in `gatsby-plugin-theme-ui`](https://github.com/c-kirkeby/portfolio-gatsby/blob/master/packages/gatsby-theme-portfolio-blog/src/gatsby-plugin-theme-ui/index.js#L6).

**Additional context**
The site will generally flash for a second and then the screen goes white.

**Comment by alexpermyakov** [comment]
Websites work very differently when you block site data (cookies, localStorage): 
Examples:
- facebook displays an error message: cookies disabled, cannot authorize
- techcrunch simplies the page (probably because personalization is disabled)

But with theme-ui, it breaks the site and displays WSoD 😨 

This is not really known localStorage issue, but it throws an exception in this case:
https://www.w3.org/TR/webstorage/#dom-localstorage

_The user agent may throw a SecurityError exception and abort these steps instead of returning a Storage object if the request violates a policy decision (e.g. if the user agent is configured to not allow the page to persist data)._

**Comment by jxnblk** [comment]
@c-kirkeby I believe this was fixed with #576, but feel free to reopen if you're still noticing issues after updating
